tests/test_model.py

```diff
@@ -106,6 +107,61 @@ def test_last_modified(self):
         self.assertLessEqual(org.last_modified, after_modified_dt)
 
 
+class TestGroup(TransactionTestCase):
+    """Unit tests for Group class"""
+
+    def test_unique_groups(self):
+        """Check whether groups are unique for organization"""
+
+        with self.assertRaisesRegex(IntegrityError, DUPLICATE_CHECK_ERROR):
+          org = Organization.objects.create(name='Example')
+          group = Group.add_root(name='subGroup1', organization=org)
+          group.add_child(name='subGroup1', organization=org)
+
+
+    def test_null_organizations(self):
+        """Check if groups can be created without organizations"""
+
+        group = Group.add_root(name='subGroup1')
+        self.assertIsInstance(group,Group)
```

Comment:
Have you tried to create a second instance with the same name and the organization as null? I think it will allow you when that shouldn't be the case.

Reply:
yes, just caught this. I should have put in a unit test for this one. I think I can override the models validate_unique or clean function to catch this.  What are your thoughts?

Reply:
@sduenas I have added a `validate_unique` function to the model and called `group.full_clean()` before saving the objects in `add_group` function in `db.py`. It catches this error. Do you think that will suffice? I will add sufficient tests as well.

Reply:
We need a test for that case to check that it doesn't let it create two groups/teams with the same name and org = NULL.

Reply:
Done :+1